BootloaderCommonPkg/BootloaderCommonPkg.dec

```diff
@@ -50,6 +50,8 @@
   gPlatformCommonLibTokenSpaceGuid.PcdHeciLibId              |          5 |  UINT8 | 0x20000106
   gPlatformCommonLibTokenSpaceGuid.PcdMmcTuningLibId         |          6 |  UINT8 | 0x20000107
 
+  gPlatformCommonLibTokenSpaceGuid.PcdContainerMaxNumber     |          8 | UINT32 | 0x20000120
```

Comment:
I am still thinking it makes code and usage a little complicated by having multiple container support.
a) too many layer: Multiple containers -> multiple regions in container -> one or multiple files in image in region.
b) Caller need to know containers beside region name
c) If different container is verified by same key,  will it bring security concern? if verified by different key, need add multiple slot in hashstore for different container.

Reply:
There are use cases for multiple containers.  
1. Component in containers might have different level of importance.  If A and B in the same container, but A needs redundancy, B does not.  It will cause issue.  If we put container in redundant region, then B will be duplicated and waste flash space.  If we put container in non-redundant region, then A will have no redundancy and break redundant boot.
2. Container might need to be updated or created by different owners.  Grouping all components in one container makes this difficult to support it.  

It is expected to use container signature to load a component.  Just like loading a file, you have to know which directory.

I did not see issue to sign different container header with the same key.  This is only for header signing. Each component inside can still have its own key.

Reply:
Looks one container could support the use cases you listed.
1. For the first use case to support redundancy. If A is boot critical (e.g. CfgData), we could put it in regular redundancy region managed by flashmap. If A is not boot critical, we could put A1 and A2 in container. 
2. We will provide tool to generate capsule image for firmwareupdate. From user viewpoint, it is transparent. they only need provide image components.
Given all the containers are signed by same key, and if user could generate and update container. It means one user of a container could update the other containers. In this case, it looks there is not much help by providing different key to sign components inside a container since the container including the component key could be updated.

Reply:
That will need to handle different layout.   And if you have more components like that, you will be running out of control pretty soon.  I didn't see why multiple containers are complicated.  It is almost the same as support single container.